File: Desktop/src/api/heldSales.ts
import { api } from './axios';
import type { HeldSale, CreateHeldSaleInput } from '@/types/heldSale';

export const heldSalesApi = {
  list: () =>
    api.get<{ data: HeldSale[] }>('/client/held-sales').then((r) => r.data.data),

  create: (payload: CreateHeldSaleInput) =>
    api
      .post<{ data: HeldSale }>('/client/held-sales', payload)
      .then((r) => r.data.data),

  resume: (id: string) =>
    api
      .post<{ data: HeldSale }>(`/client/held-sales/${id}/resume`)
      .then((r) => r.data.data),

  remove: (id: string) =>
    api.delete(`/client/held-sales/${id}`).then((r) => r.data),
};

File: Android/client/lib/api/heldSales.dart
// ignore_for_file: file_names

import 'api_client.dart';

class HeldSalesApi {
  static final _api = BizOsApiClient.instance;

  static Future<Response<dynamic>> list({Map<String, dynamic>? query}) =>
      _api.get('/client/held-sales', query: query);
  static Future<Response<dynamic>> create(Map<String, dynamic> data) =>
      _api.post('/client/held-sales', data: data);
  static Future<Response<dynamic>> resume(String id) =>
      _api.post('/client/held-sales/$id/resume');
  static Future<Response<dynamic>> remove(String id) =>
      _api.delete('/client/held-sales/$id');
}
